data_structures/queueInC.c

```diff
@@ -0,0 +1,140 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include <string.h>
+
+typedef struct link{
+    int element;
+    struct link *next;
+} Link;
+
+typedef struct{
+    Link *front;
+    Link *rear;
+    int size;
+} Queue;
+
+Link *create_Link(int it, Link *nextval){
+    Link *n;
+    n = (Link*) malloc(1*sizeof(Link));
+    n->element = it;
+    n->next = nextval;
+    return n;
+}
+
+Link *create_link(Link *nextval){
+    Link *n;
+    n = (Link*) malloc(1*sizeof(Link));
+    n->next = nextval;
+    return n;
+}
+
+Queue *create_queue(){
+    Queue *q;
+    q = (Queue*) malloc(1*sizeof(Queue));
+    (*q).rear = create_link(NULL);
+    q->front = q->rear;
+    q->size = 0;
+    return q;
+}
+
+void clear(Queue *q){
+    q->rear = q->front;
+    Link *tmp;
+    while(q->rear != NULL) {
+        tmp = q->rear;
+        q->rear = q->rear->next;
+        free(tmp);
+    }
+    q->size = 0;
+    free(q);
+}
+
+void enqueue(Queue *q, int it){
+    q->rear->next = create_Link(it, NULL);
+    q->rear = q->rear->next;
+    q->size++;
+}
+
+int dequeue(Queue *q){
+    Link *temp;
+    if(q->size == 0){
+        return -1;
+    }
+    int it = q->front->next->element;
+    temp = q->front->next;
+    q->front->next = q->front->next->next;
+    if(q->front->next == NULL){
+        q->rear = q->front;
+    }
+    free(temp);
+    q->size--;
+    return it;
+}
+
+int frontValue(Queue *q){
+    if(q->size == 0){
+        return -1;
+    }
+    return q->front->next->element;
+}
+
+int lenght(Queue *q){
+    return q->size;
+}
+
+#define max 1000
+
+int main(){
+    
+    Queue *left, *right;
+    int i, c, cont, flag = 1, viagem = 0;
+    int l, m, it;
+    int actual, total = 0, total2 = 0;
+    char side[max];
+
+    scanf("%d", &c);
+    //printf("Casos Teste %d\n", c);
+
+    while(cont < c){
```

Comment:
``cont`` is never initialized

Reply:
For sure, this is a huge mistake. I corrected now this issue initializing my cont as zero.